changelogs/unreleased/gh-7650-forbid-same-names-for-constraints-and-fkeys.md

```diff
@@ -0,0 +1,4 @@
+## feature/core
+
+* It is now forbidden for a check constraint and a foreign key of each field
+  (or entire tuple) to have the same name (gh-7650).
```

Comment:
I'm not sure this change is 100% safe because there may be installations that use duplicate names for some reason. It may be worth disabling the checks during recovery or introduce a `compat` module option for it. Up to @alyapunov.